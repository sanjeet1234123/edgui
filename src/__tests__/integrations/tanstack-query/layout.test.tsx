// @ts-nocheck
/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import LayoutAddition from '@/integrations/tanstack-query/layout'

// Mock the Devtools to avoid side effects and check rendering
jest.mock('@tanstack/react-query-devtools', () => ({
  ReactQueryDevtools: jest.fn(({ buttonPosition }) => (
    <div data-testid="react-query-devtools" data-position={buttonPosition} />
  )),
}))

describe('LayoutAddition', () => {
  it('renders ReactQueryDevtools with correct button position', () => {
    render(<LayoutAddition />)
    const devtools = screen.getByTestId('react-query-devtools')
    expect(devtools).toBeInTheDocument()
    expect(devtools).toHaveAttribute('data-position', 'bottom-right')
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<LayoutAddition />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders only one Devtools instance', () => {
    render(<LayoutAddition />)
    expect(screen.getAllByTestId('react-query-devtools')).toHaveLength(1)
  })
})
