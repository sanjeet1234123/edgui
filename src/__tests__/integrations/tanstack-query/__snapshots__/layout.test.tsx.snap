// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutAddition matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-position="bottom-right"
    data-testid="react-query-devtools"
  />
</DocumentFragment>
`;
